torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -110,21 +111,23 @@ def __init__(self,
         self._enabled = True
         self._seed = None
         self._rng = random.Random()
+        # TODO: Using the initial RNG state until the full graph of DataPipe
+        #       can be serialized in the middle of iteration
+        self._rng_state = self._rng.getstate()
```

Comment:
I am saving the initial `rng_state` whenever we seeded the `rng` rather than the state in the middle of iteration. This is the same as `self._buffer` is never saved.

For proper serialization, we should save both `buffer` and `rng_state` when `__getstate__` is called.

Reply:
I see now. That makes sense. Thanks!